tests/feature_extraction_test.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+from translation.feature_extraction_module import FeatureExtractionConfig, FeatureExtractionModule, LauncherConfig
+from unittest.mock import patch, MagicMock
+
+class TestFeatureExtractionModule(unittest.TestCase):
+    def setUp(self):
+        # Mock the configuration for the FeatureExtractionModule
+        self.config = FeatureExtractionConfig(
+            data_dir="MOCK_dataset",
+            pretrained_model_path="signhiera_mock.pth", 
```

Comment:
Ideally, the test would also download this mock encoder for the user instead of expecting it to be downloaded in a specific place first